lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -546,6 +548,11 @@ void MooreToCorePass::runOnOperation() {
   MLIRContext &context = getContext();
   ModuleOp module = getOperation();
 
+  auto pm = PassManager::on<ModuleOp>(&context);
+  pm.addNestedPass<moore::SVModuleOp>(moore::createDeleteLocalVarPass());
+  if (failed(pm.run(module)))
+    return signalPassFailure();
+
```

Comment:
I would suggest to not nest this pass inside `MooreToCore`. You want passes and conversions to be more or less independent pieces of work, and then write a tool such as firtool, circt-verilog, or arcilator to create a pass manager and line up the passes you need to execute.

In this case here, I'd suggest to add the pass manager and `DeleteLocalVarPass` to the circt-verilog tool. Maybe also with an output format option, similar to firtool. For example:
- `--parse-only` to just run the `ImportVerilog` pass and emit the resulting Moore dialect IR
- `--ir-moore` to additionally run the entire pass manager (just the DLV pass for now) to just before `MooreToCore` conversion, and emit the resulting Moore dialect IR
- `--ir-hw` to additionally run the `MooreToCore` conversion and emit the resulting core dialect IR

That will keep `MooreToCore` a pure conversion that does only one specific thing, and which requires the IR to be in a specific shape (like all local vars to be deleted), and make `circt-verilog` the tool to go from Slang (or later also Moore IR) to the core dialects. Similar to how firtool works.

Reply:
I'll try and complete it.

Reply:
I move the `DeleteLocalVarPass` into the circt-verilog tool. `--parse-only` has been used by input(LoweringMode). I added three options: `--unspecified`: like `--parse-only`, `--ir-moore`, and `--ir-hw`. Should I merge LoweringMode and OutputFormatKind?